test/index.test.js

```diff
@@ -911,5 +912,32 @@ describe('mongoose module:', function() {
       const m = new mongoose.Mongoose();
       assert.deepEqual(await m.syncIndexes(), {});
     });
+    it('Allows for the removal of indexes via string or object (gh-11547)', async function() {
```

Comment:
@IslandRhythms 
We need to cover the case where a schema is using another schema with indexes.
For example
```js
const productSchema = new Schema({ name: String });
productSchema.index({ name: 1 });

const orderSchema = new Schema({
  products: [productSchema]
});
```

Reply:
At first glance, I'm opposed to having `orderSchema.clearIndexes()` clear all indexes on child schemas, at least by default. This can cause surprises if you use `productSchema` in multiple places.

Reply:
We currently don't clone child schemas. In your example, `orderSchema` contains a reference to `productSchema`, so if `orderSchema.clearIndexes()` cleared indexes on child schemas, it would clear `productSchema`'s indexes as well. That behavior isn't ideal, but given that we don't clone child schemas we shouldn't make `clearIndexes()` clear child schema indexes by default. We can add an option though.